libnetwork/resolver.go

```diff
@@ -75,7 +76,7 @@ type Resolver struct {
 	tcpListen     *net.TCPListener
 	err           error
 	listenAddress string
-	proxyDNS      bool
+	proxyDNS      *atomic.Bool
```

Comment:
does `Resolver` need to be copyable?
```suggestion
	proxyDNS      atomic.Bool
```